### ISSUE TITLE
Maturin does not see [tool.poetry] synonyms in pyproject.toml (PEP 621)

### ISSUE THREAD

**Issue by tmtenbrink** [PyO3/maturin#632]
Thanks for this great project, it greatly simplified packaging some Rust bindings! 

Please provide the following information: 
 * Your python version (`python -V`): 3.9.5
 * Your pip version (`pip -V`): 21.2.4
 * The version of the bindings you're using, if any (e.g. `pyo3`, `rust-cpython` or `cffi`): pyo3
 * Does `cargo build` work? Yes

On my [tmtenbrink/rustfrc](https://github.com/tmtenbrink/rustfrc) repository, I was originally using a dual [project] as well as [tool.poetry] section as I originally believed some metadata was only available in [project] to be read by PyPI and pip. However, after studying [PEP 621](https://www.python.org/dev/peps/pep-0621/) and the python poetry docs, it seems like [tool.poetry] and [project] information are synonyms. As such, it shouldn't be necessary to both have [project] and [tool.poetry] sections. However, after removing the [project] sections, the metadata is no longer present in the built wheels. As such, I believe this is a problem with maturin not adding the [tool.poetry] information to the built wheels metadata. This also means that dependencies are not transferred to the wheel metadata and they are not installed when not using pip.

In short: maturin does not see the [tool.poetry] information as synonyms. I.e. [tool.poetry.urls] is not seen as a synonym of [urls] under a [project], which I believe it should based on PEP 621.

**Comment by messense** [comment]
I don't think PEP 621 mandates that build backend MUST support synonyms, if it does please paste the relevant texts here.

PEP 621 is designed to standardize how to store Python project metadata, what I expect is that Poetry should support the new `[project]` table instead of require other tools to be compatible with Poetry's specification.

**Comment by messense** [comment]
Poetry does have a tracking issue for this https://github.com/python-poetry/poetry/issues/3332
